parsers/AX.py

```diff
@@ -75,15 +75,22 @@ def formated_data(
     """Format data to Electricity Map standards."""
     data_list = fetch_data(session, logger)
     data_list.reverse()
-    date = datetime.now(timezone(TIME_ZONE)).replace(
+    date_time = datetime.now(timezone(TIME_ZONE))
+    date = date_time.replace(
         hour=int(data_list[0]["time"].split(":")[0]),
         minute=int(data_list[0]["time"].split(":")[1]),
         second=0,
         microsecond=0,
     )
     return_list = []
     for data in data_list:
-        corrected_date = date - timedelta(minutes=15 * data_list.index(data))
+        index = data_list.index(data)
+        corrected_date = date - timedelta(minutes=15 * index)
+        if corrected_date > (date_time - timedelta(minutes=15 * index)):
+            raise ParserException(
```

Comment:
Looking at this code, instead of raising a `ParserExcepion`, wouldn't it be better to correct the date to yesterday? 

Probably something along the lines of 

    date = date_time.replace(
        day=<yesterday>